app/controllers/api/v2/courses_controller.rb

```diff
@@ -56,6 +56,24 @@ def update
         enrichment.assign_attributes(update_params)
         enrichment.save
 
+        # todo: something more canonical
+        # todo: validate sites actually belong to this org
+        desired_site_ids = params["_jsonapi"]["data"]["relationships"]["sites"]["data"].map{|s| s["id"].to_i}
+
+        existing_site_ids = @course.sites.map{|s| s["id"]}
+
+        to_add = desired_site_ids - existing_site_ids
+        to_add.each do |id|
```

Comment:
this is basically 
select one drop one looped....
instead of `.find(id)` `.find(to_add)`?